rust/src/interpreter/stack.rs

```diff
@@ -6,11 +6,7 @@ use crate::types::u256;
 pub struct Stack(Vec<u256>);
 
 impl Stack {
-    pub fn new() -> Self {
-        Self(Vec::new())
-    }
-
-    pub fn from_vec(inner: Vec<u256>) -> Self {
+    pub fn new(inner: Vec<u256>) -> Self {
```

Comment:
I liked the empty constructor :(

I think the reason why you removed it is that you got tired of repeating code. This has to do with ergonomics. 
IMO, new (for empty) and new_with(slice)  would be less code (no vector). Yes, they hide a vector copy, data is so small that it could be optimized away, and overhead in tests will be negligible, I would prefer ergonomics to premature optimization in this case.

Reply:
I removed it because after some refactoring of the next PRs the empty constructor was no longer needed. And in the tests I think it is cleaner to provide an empty vector to the (few) cases where an empty Stack is needed than to use a different constructor.